src/fb5.js

```diff
@@ -234,18 +239,46 @@ function setupPageObserver() {
 
 // cleanup
 window.addEventListener("beforeunload", () => {
-  pageObserver.disconnect();
+  cleanupPageObserver();
 
   if (feedObserver !== null) {
-    feedObserver.disconnect();
-    feedObserver = null;
+    cleanupFeedObserver();
   }
 
   if (watchObserver !== null) {
-    watchObserver.disconnect();
-    watchObserver = null;
+    cleanupWatchObserver();
   }
 });
+function cleanupPageObserver() {
+  if (pageObserver === null) return;
+  pageObserver.__observed &&
+    delete feedObserver.__observed.dataset.adblockObserved &&
+    delete feedObserver.__observed;
```

Comment:
nit: I'm not a fan for shorthand condition. Is there a reason why we should prefer this over a full `if`?

Reply:
changed. 
This particular block had also a typo where I didn't change the feedObserver to pageObserver when I copy-pasted that block. It's also fixed